tests/integration/formatter/formatter_test.py

```diff
@@ -0,0 +1,157 @@
+import os
+from typing import List, Dict
+from pathlib import Path
+
+import pytest
+
+from tests.integration.protostar_fixture import ProtostarFixture
+from tests.data.contracts import (
+    BROKEN_CONTRACT,
+    FORMATTED_CONTRACT,
+    UNFORMATTED_CONTRACT,
+)
+
+
+@pytest.fixture(autouse=True, scope="function")
+def setup_function(protostar: ProtostarFixture):
+    protostar.init_sync()
+
+    cwd = Path().resolve()
+    os.chdir(protostar.get_project_root_path())
+
+    protostar.create_files(
+        {
+            "to_format/formatted.cairo": FORMATTED_CONTRACT,
+            "to_format/unformatted1.cairo": UNFORMATTED_CONTRACT,
+            "to_format/unformatted2.cairo": UNFORMATTED_CONTRACT,
+            "to_format/broken.cairo": BROKEN_CONTRACT,
+        }
+    )
+    yield
+
+    os.chdir(cwd)
```

Comment:
`ProtostarFixture` needs refactoring, but this is a separate task.

```suggestion
    project_root = protostar.init_sync()
    with cwd(project_root):
       protostar.create_files(...)
       # ...
       yield
```